src/widgets/input/combineInput.js

```diff
@@ -0,0 +1,533 @@
+      <CombineText disabled={disabled} themeProps={themeProps}>
+        {text}
+      </CombineText>
+    ) : null;
+  }
+
+  getDisplayIcon(position: _InnerPositionType, iconClass: string, iconPosition: IconPositionType) {
+    const { getPartOfThemeHocProps, disabled, size } = this.props;
+    const { theme, viewClass } = getPartOfThemeHocProps(`${position}Icon`);
+
+    const theIconPosition = checkValue(iconPosition, 'after') ? 'left' : 'right';
+    const iconDefaultTheme = {
+      [viewClass]: {
+        normal: {
+          color: blackColor,
+          getThemeMeta(themeMeta) {
+            const { fontSize, font: { size: innerFontSize } = {} } = themeMeta;
+            const theSize = innerFontSize || fontSize || getIconSize(size);
+            return {
+              fontSize: theSize,
+              margin: {
+                [theIconPosition]: paddingToText,
+              },
+            };
+          },
+        },
+        disabled: {
+          color: disableTextColor,
+        },
+      },
+    };
+    const theTheme = deepMerge(iconDefaultTheme, theme);
+    return iconClass !== '' ? (
+      <Icon
+        size={size}
+        iconPosition={iconPosition}
+        disabled={disabled}
+        singleTheme
+        iconClass={iconClass}
+        viewClass={viewClass}
+        theme={theTheme}
+      />
+    ) : null;
+  }
+
+  getDisplayRender(position: _InnerPositionType) {
+    return checkValue(position, _BeforeType)
+      ? this.getBeforeRenderContent()
+      : checkValue(position, _AfterType)
+      ? this.getAfterRenderContent()
+      : '';
+  }
+
+  getAfterRenderContent() {
+    const {
+      afterType,
+      afterRenderValue,
+      afterIconPosition,
+      afterIconClass = defaultIconClass,
+    } = this.props;
+    return checkValue(afterType, 'display')
+      ? this.getDisplayInnerContent(_AfterType, afterRenderValue, afterIconClass, afterIconPosition)
+      : checkValue(afterType, 'custom')
+      ? afterRenderValue
+      : '';
+  }
+
+  getBeforeRenderContent() {
+    const {
+      beforeType,
+      beforeRenderValue,
+      beforeIconPosition,
+      beforeIconClass = defaultIconClass,
+    } = this.props;
+
+    return checkValue(beforeType, 'display')
+      ? this.getDisplayInnerContent(
+          _BeforeType,
+          beforeRenderValue,
+          beforeIconClass,
+          beforeIconPosition
+        )
+      : checkValue(beforeType, 'custom')
+      ? beforeRenderValue
+      : '';
+  }
+
+  getDisplayInnerContent(
+    position: _InnerPositionType,
+    renderValue: string | React.ReactNode,
+    iconClass: string,
+    iconPosition: IconPositionType
+  ) {
+    if (checkValue(iconPosition, 'after')) {
+      return [
+        this.getDisplayText(position, renderValue),
+        this.getDisplayIcon(position, iconClass, iconPosition),
+      ];
+    }
+    return [
+      this.getDisplayIcon(position, iconClass, iconPosition),
+      this.getDisplayText(position, renderValue),
+    ];
```

Comment:
const a =  this.getDisplayText(position, renderValue);
const b =  this.getDisplayIcon(position, iconClass, iconPosition);
if(){
return [a,b]
}
return [b,a]